### PR TITLE
Improve robustness computations of log3

### PR DESCRIPTION
**Pull request by jcarpent**

### CODE REVIEW

File: src/spatial/log.hxx [hunk 1]
```diff
@@ -40,9 +40,9 @@ namespace pinocchio
         // using explicit formula. However, the precision of this method
         // is the square root of the precision with the antisymmetric
         // method (Nominal case).
-        const Scalar cphi = cos(theta - PI_value);
+        const Scalar cphi = -(tr - Scalar(1))/Scalar(2);
```
Comment (jmirabel) [inline src/spatial/log.hxx]: Mind that *tr* was not threshold-ed, thus *cphi* may have a norm greater than 1. No idea whether this can have an impact.

Comment (jcarpent) [reply src/spatial/log.hxx:43]: For these computations, tr is in for sure ]-1,3[

Comment (jmirabel) [reply src/spatial/log.hxx:43]: It is far from 3 but close to -1 and it can be slightly below -1 because of numerical imprecision.

Comment (jmirabel) [reply src/spatial/log.hxx:43]: If tr < -1, then line 30 sets theta to Pi so the condition of line 37 is true.

**Pull request merged**